hooks/990-ensure-uids-gids.chroot

```diff
@@ -31,8 +31,8 @@ irc:x:39:39:ircd:/var/run/ircd:/usr/sbin/nologin
 gnats:x:41:41:Gnats Bug-Reporting System (admin):/var/lib/gnats:/usr/sbin/nologin
 nobody:x:65534:65534:nobody:/nonexistent:/usr/sbin/nologin
 _apt:x:100:65534::/nonexistent:/usr/sbin/nologin
-systemd-network:x:101:103:systemd Network Management,,,:/run/systemd/netif:/usr/sbin/nologin
-systemd-resolve:x:102:104:systemd Resolver,,,:/run/systemd/resolve:/usr/sbin/nologin
+systemd-network:x:101:102:systemd Network Management,,,:/run/systemd/netif:/usr/sbin/nologin
```

Comment:
Hm, why did the UIDs change here? Also, will they still be compatible with core16? If not, I think we need to create our own passwd (similar to what we do on the phone/core16).

Reply:
Hey! We do that already actually! The UIDs change frequently whenever new packages are installed, due to package installation ordering it seems. The only thing that happened here is that the input group moved to a later slot than it was, just because of this one new dependency installed that seemed to influence which parts were triggered first. We could try forcing the ordering by doing separate apt install operations, but since we already had this problem before we're just making sure that the groups and users are 'sane' and then create our own versions of passwd, group, shadow and gshadow (that's being done later in the hook). I already tried forcing apt installation ordering and it looked horrible.

Reply:
Aha, sorry. I misread that this change is just in the check code. But I still wonder if we do the right thing here. It seems like if "systemd-network" would create any files on the FS those would be written as "101:103" and then we change the passwd/group and "systemd-network" is now "101:102" so these files have wrong group ownership on disk?. 

Unless I miss something maybe we need a two stage approach: create the passwd/group early, i.e. before we install packages. This will mean that all the adduser calls in the postinst will become no-ops. Then do a "late" check to ensure that things fails when new users/groups getc added by apt/postinsts. If this happens we just add those to our static script and the uid/gid map will be static and fine again.

Does that make sense?